src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/DropNamespaceCommand.java

```diff
@@ -0,0 +1,21 @@
+package io.stargate.sgv2.jsonapi.api.model.command.impl;
+
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import io.stargate.sgv2.jsonapi.api.model.command.GeneralCommand;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Pattern;
+import javax.validation.constraints.Size;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+
+@Schema(description = "Command that deletes a namespace.")
+@JsonTypeName("dropNamespace")
+public record DropNamespaceCommand(
+    @NotNull
+        @Pattern(regexp = "[a-zA-Z][a-zA-Z0-9_]*")
+        @Pattern(regexp = "^(?!system$).*")
```

Comment:
Seems wrong considering these are all system keyspaces:

```
 keyspace_name      | durable_writes | graph_engine | replication
--------------------+----------------+--------------+-------------------------------------------------------------------------------------
        system_auth |           True |         null | {'class': 'org.apache.cassandra.locator.SimpleStrategy', 'replication_factor': '1'}
      system_schema |           True |         null |                             {'class': 'org.apache.cassandra.locator.LocalStrategy'}
   dse_system_local |           True |         null |                             {'class': 'org.apache.cassandra.locator.LocalStrategy'}
         dse_system |           True |         null |                        {'class': 'org.apache.cassandra.locator.EverywhereStrategy'}
         dse_leases |           True |         null | {'class': 'org.apache.cassandra.locator.SimpleStrategy', 'replication_factor': '1'}
         solr_admin |           True |         null |                        {'class': 'org.apache.cassandra.locator.EverywhereStrategy'}
       dse_insights |           True |         null |                        {'class': 'org.apache.cassandra.locator.EverywhereStrategy'}
 dse_insights_local |           True |         null |                             {'class': 'org.apache.cassandra.locator.LocalStrategy'}
 system_distributed |           True |         null | {'class': 'org.apache.cassandra.locator.SimpleStrategy', 'replication_factor': '3'}
     system_backups |           True |         null |                        {'class': 'org.apache.cassandra.locator.EverywhereStrategy'}
             system |           True |         null |                             {'class': 'org.apache.cassandra.locator.LocalStrategy'}
           dse_perf |           True |         null | {'class': 'org.apache.cassandra.locator.SimpleStrategy', 'replication_factor': '1'}
 data_endpoint_auth |           True |         null | {'class': 'org.apache.cassandra.locator.SimpleStrategy', 'replication_factor': '1'}
      system_traces |           True |         null | {'class': 'org.apache.cassandra.locator.SimpleStrategy', 'replication_factor': '2'}
       dse_security |           True |         null | {'class': 'org.apache.cassandra.locator.SimpleStrategy', 'replication_factor': '1'}
```

Anybody with idea how to solve this?

Reply:
Do we enforce similar restriction on other APIs? 

One way to protect against this would be through permissions.

Reply:
We don't enforce in the other apis..

Reply:
I think we need to enforce on data_endpoint_auth keyspace, since it's created by stargate and storage may not restrict it.

Reply:
I think I will remove any limitations on the keyspace name here. As @tatu-at-datastax pointed, the jsonapi should not be responsible in protecting which keyspaces can be deleted or not.

Even in case of the `data_endpoint_auth` keyspace, it should be the responsibility up in the chain (coordinator).

Furthermore, as @jeffreyscarpenter mentioned yesterday, there is no metadata for the keyspaces in cassandra, so there is no way for us to "label" a keyspace during a creation process and know exatcly which ones are created with the jsonapi.

The only additional idea which comes to my mind (and is out of the scope for this PR), would be to allow deletion of a kespace only if it empty (no table) or all tables are schema-wise the jsonapi tables. Similar technique, we can use in the `findNamespaces` and list namesapces only if they are empty or at least one table is schema-wise a jsonapi table. Would like to hear opinions about this?

Reply:
I like the idea of only allowing dropping if no tables exist in the namespace.

Question on listing is interesting... I can see benefits of only listing only ones that have at least one Collection, although it might be problem if user creates namespace and wants to verify it exists before creating collection.
Maybe there would be an option for "show empty namespaces".

Reply:
@tatu-at-datastax Yea I included the empty keyspace as well:

* for listing: no tables or at least one table is jsonapi schema
* for dropping: no tables or all are jsonapi schema